### ISSUE TITLE
[2024-12-18] Refactor page tree and add it to to statistics as a collapsible box

### ISSUE THREAD

**Issue by JoeyStk** [digitalfabrik/integreat-cms#3130]
### Motivation

We want to add the access statistics for individual pages to our statistics. For this we need a collapsible box containing a replicated page tree.

### Proposed Solution

Refactor page tree first and then add one instance of the page tree to the statistics page as a collapsible box.

### Alternatives

None

### User Story

As a user of the CMS I want to see the number of accesses to individual pages in a hierarchical way so that I know which content is more generally relevant for the majority of users.

### Additional Context

This part of our meta issue #1097

### Design Requirements

Design proposal (v3):
https://www.figma.com/file/6U7R7Xj4wL7sbjxKRmOG9D/CMS-Project?node-id=1179-830

ToDos:
- [ ] Refactor page list view
- [ ] New template for `_generic_page_tree`
- [ ] `pages_page_trree`
- [ ] `statistics_page_tree`

**Comment by charludo** [comment]
Most of the issues linked in #1097 are blocked by this issue. I would therefore like to get started on it, but tbh I am kinda lost on what/how much needs to be refactored.

Is it enough to make the page tree "work" for the purpose of showing the new statistics? Or do we actually want a ground-up, modular rebuild?

Both are certainly possible, but I am struggling to see if there is a good middle ground here.

**Comment by JoeyStk** [comment]
> Is it enough to make the page tree "work" for the purpose of showing the new statistics? Or do we actually want a ground-up, modular rebuild?

As I remember the idea was to "refactor" so much, that we can reuse as much as possible. Therefore I'm leaning towards the first. But @david-venhoff is there maybe something else you also would like to see get done with this issue? :)

**Comment by david-venhoff** [comment]
No, I also think that it is enough to get the page tree to work for statistics :)

**Comment by jarlhengstmengel** [comment]
'@JoeyStk i merged `archived_page_tree.html` into `page_tree.html`. The page tree and the archived page tree deal with their rows quite differently, so the archived drop down of the rows has still some errors and I haven't merged the respective row templates yet

**Comment by jarlhengstmengel** [comment]
Should we also implement a generic tree node? Would that make sense?

**Comment by JoeyStk** [comment]
@jarlhengstmengel 
Remaining tasks:
- [ x ] Fix issue with icon
- [ x ] Fix issue with strange spacing
- [ ] Fix issue with page tree
- [ ] Change page tree to be not editable in statistics (as for archived pages)
- [ ] Fix issue to select a time frame
- [ ] Fix issue to select language

**Comment by jarlhengstmengel** [comment]
Regarding the refactoring of the archived page tree, the idea  came up to show the (not archived) ancestors of a archived page so that users get a better overview where in the page tree the archived page was located. I think that could be helpful information when a page should be restored. Is that something we would like to do @JoeyStk?

**Comment by jarlhengstmengel** [comment]
> > Regarding the refactoring of the archived page tree, the idea came up to show the (not archived) ancestors of a archived page so that users get a better overview where in the page tree the archived page was located. I think that could be helpful information when a page should be restored. Is that something we would like to do @JoeyStk?
> 
> I have wondered about something similar in the past. I assume we would need to do a good job with it and stay in touch with ui-ux during the process, but all in all I like the idea to improve the current view

@osmers @nikolahoff what do you think? Currently it's just a plain list of archived pages without any special structure. In the current state of refactoring, the hierarchy between a archived page and it's implicitly archived children is visible, but without the possibility of collapsing or expanding as in the page tree since that turned out to be quite complicated to realize. With showing the non archived parents, the dropdown should be a lot easier to implement, but I think that a good visual distinction would be needed

**Comment by nikolahoff** [comment]
@jarlhengstmengel can you provide a link or screenshot? thx

**Comment by jarlhengstmengel** [comment]
@nikolahoff here the current state of how archived pages look in the refactoring
![image](https://github.com/user-attachments/assets/96ec2a8a-d129-4e3b-bcd5-92a30e08840c)

**Comment by jarlhengstmengel** [comment]
@nikolahoff a question came up during the review of the refactoring of the page tree. In the statistics and the archived pages view, drag and drop of pages is disabled to prevent accidentally changing the page order. At the moment, a message is shown that drag and drop is disabled but it's the only visual clue while the symbol signaling drag and drop is active is still visible. Do you have a idea to make it a bit more consistent/clear?

![Image](https://github.com/user-attachments/assets/45db8988-2c91-44f9-a794-a69ac4416bf3)

**Comment by jarlhengstmengel** [comment]
> [@nikolahoff](https://github.com/nikolahoff) a question came up during the review of the refactoring of the page tree. In the statistics and the archived pages view, drag and drop of pages is disabled to prevent accidentally changing the page order. At the moment, a message is shown that drag and drop is disabled but it's the only visual clue while the symbol signaling drag and drop is active is still visible. Do you have a idea to make it a bit more consistent/clear?
> 
> ![Image](https://github.com/user-attachments/assets/45db8988-2c91-44f9-a794-a69ac4416bf3)

@nikolahoff I think I solved it quite close to the initial [design](https://www.figma.com/design/6U7R7Xj4wL7sbjxKRmOG9D/CMS-Project?node-id=533-3487&p=f). 

![Image](https://github.com/user-attachments/assets/cacf05eb-4257-4a6c-8b8e-926a8d10d6ce)

**Comment by nikolahoff** [comment]
@jarlhengstmengel yes, when the arrow is not shown it doesn't look like moving is possible. But would there be a reason someone would want to move pages in the statistics area?

**Comment by charludo** [comment]
> @jarlhengstmengel yes, when the arrow is not shown it doesn't look like moving is possible. But would there be a reason someone would want to move pages in the statistics area?

Sorry, but isn't it confusing for users to be presented with UI elements that have a clearly defined function in other parts of the CMS, but are useless / do not make sense in this context? 😅

**Comment by nikolahoff** [comment]
@charludo I don't understand what you mean... Which UI elements do you mean? As I understood Jarl changed it to like it is shown in the 2. screen where the arrows are not shown anymore.

**Comment by jarlhengstmengel** [comment]
> [@jarlhengstmengel](https://github.com/jarlhengstmengel) yes, when the arrow is not shown it doesn't look like moving is possible. But would there be a reason someone would want to move pages in the statistics area?

No, there is probably no reason to be able to move pages in the statistics area. Then I leave it like in the 2. Screenshot. Thanks for the feedback :)
